src/Microsoft.Extensions.DependencyInjection.Abstractions/IRequiredServiceAware.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Implementation which gets invoked on the <see cref="IServiceProvider"/> when
+    /// <see cref="ServiceProviderExtensions.GetRequiredService{T}(IServiceProvider)"/> or
+    /// <see cref="ServiceProviderExtensions.GetRequiredService(IServiceProvider, Type)"/> is called.
+    /// </summary>
+    public interface IRequiredServiceAware
```

Comment:
And `IRequiredServiceProvider` doesn't? I'm really not a fan of the `IProvideServices`, `ICreateLoggers`, `IActivateControllers` naming, compared to `IServiceProvider`, `ILoggerFactory` and `IControllerActivator`. It would be nice if it were consistent. Just my 2¢ :smile: